packages/smooth_app/lib/pages/product/common/product_list_add_button.dart

```diff
@@ -0,0 +1,30 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_gen/gen_l10n/app_localizations.dart';
+import 'package:smooth_app/pages/product/common/product_query_page_helper.dart';
+import 'package:smooth_app/pages/product/common/smooth_chip.dart';
+
+/// "Add button" related to a user product list type
+class ProductListAddButton extends StatelessWidget {
+  const ProductListAddButton({
+    required this.onPressed,
+    required this.onlyIcon,
+    required this.productListType,
+  });
+
+  final Function onPressed;
+  final bool onlyIcon;
+  final String productListType;
```

Comment:
A general question, wouldn't it be easier to take the correct enum value from the string as early as possible and then pass this on?

Reply:
@M123-dev At some point we need `String`s because we store the product list type in the database.
`enum` would make sense, of course, but we'll still have to convert to/from `String`.
IMHO no rush for creating `enum`s.